### DISCUSSION TITLE
Reverse proxy apache2 with docker pwpush

### DISCUSSION THREAD

**Discussion by cdivriotis97** [pglombardo/PasswordPusher#835] [Q&A] [answered]
Hi everyone,

I got an issue and I am stuck for hours now.
I got an apache SSL reverse proxy behind my pwpush which is a internal machine.
I used docker-compose to set up password pusher and my config looks like this:
![image](https://user-images.githubusercontent.com/116892320/219393548-376b6e05-2b59-458f-b0c3-bf0e0adac7b1.png)

Everything looks fine when I navigate to http://< IP INTERNAL >:5100.
But with the reverse proxy, I set up the things like that:
![image](https://user-images.githubusercontent.com/116892320/219393307-0a333962-7bdd-41e4-b2d0-a855f85dee37.png)

I turned on "ProxyVia" & ProxyRequests" to "On" to test if it was the root cause but it is not the case
Password Pusher is here but when I tried to click to "Push It", nothing is happening.
The redirection doesn't seem to work:
![image](https://user-images.githubusercontent.com/116892320/219392750-b80d88f7-4714-4a7d-a376-7ee3e76dbc96.png)

I am stuck when I am using the URL with the reverse proxy to the same page when I click on"Push it" even though I added the variable PWP__OVERRIDE_BASE_URL.

What am I doing wrong here ?

Let me know if I can give you more infos.
Thank you a lot for your help !

Best regards,
Snoopy

**Comment by pglombardo** [comment, marked as answer]
Hi @snoopyx21 - could you make sure that nginx/apache2 is sending the `X-Forwarded-Host`, `X-Forwarded-Port` and `X-Forwarded-Proto` headers to Password Pusher?  This is the cause of a lot of issues.  See [this documentation](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#proxies) for a brief explanation.

That should resolve the URL/Host issues.  Could you let me know how it behaves then?

**Reply by cdivriotis97** [reply]
Hi @pglombardo 

I did some research and you're right.
I needed this in my apache config file: 
`RequestHeader set X-Forwarded-Proto https`

Thanks a lot and thank you for pwpusher. 
The tool is fantastic !

**Reply by pglombardo** [reply]
So all issues fixed?  That's excellent and thanks for the nice words!

**Reply by cdivriotis97** [reply]
Yes all fixed !
